javascript/javascript2/week1/movieApp/index.js

```diff
@@ -330,53 +69,191 @@ movieList.forEach((movie) => {
   const movieDescription = document.createElement("p");
   movieDescription.innerText = movie.description;
   detailsDiv.appendChild(movieDescription);
+
   const averageRating = document.createElement("p");
-  averageRating.innerText = `Current Rating: ${movie.rating}/5`;
+  averageRating.innerText = `Current Rating: ${movie.rating}`;
   detailsDiv.appendChild(averageRating);
 
   /* rating */
   const ratingHeader = document.createElement("h3");
   ratingHeader.innerText = "Give Ratings ";
   detailsDiv.appendChild(ratingHeader);
   const ratingsDiv = document.createElement("div");
-  ratingsDiv.className = "ratingsContainer";
-
-  let currentRating = 0;
+  ratingsDiv.setAttribute("class", "ratingsContainer");
 
-  for (let i = 0; i < 5; i++) {
+  for (let i = 5; i > 0; i--) {
     const star = document.createElement("i");
     star.className = "fa fa-star";
     star.setAttribute("data-rating", i);
     ratingsDiv.appendChild(star);
+  }
 
-    star.addEventListener("mouseenter", () => {
-      for (let j = 0; j <= i; j++) {
-        const tempIcon = ratingsDiv.children[j];
-        tempIcon.style.color = "yellow";
-      }
-    });
-
-    star.addEventListener("mouseleave", () => {
-      for (let j = 0; j < i; j++) {
-        const tempIcon = ratingsDiv.children[j];
-        tempIcon.style.color = j < movie.rating ? "yellow" : "gray";
-      }
-    });
+  const stars = ratingsDiv.querySelectorAll("i");
 
+  for (let star of stars) {
     star.addEventListener("click", () => {
-      movie.rating = i + 1;
-      currentRating = i + 1;
-      for (let j = 0; j < 5; j++) {
-        const tempIcon = ratingsDiv.children[j];
-        tempIcon.style.color = j < currentRating ? "yellow" : "gray";
+      const children = star.parentElement.children;
+      for (let child of children) {
+        if (child.getAttribute("data-clicked")) {
+          return false;
+        }
       }
-      averageRating.innerText = `Current Rating: ${movie.rating}/5`;
+
+      star.setAttribute("data-clicked", "true");
+      const rating = star.dataset.rating;
+      movie.rating = rating;
```

Comment:
You can simplify this, as anyways you call the function only once.
```suggestion
      movie.rating = star.dataset.rating;
```